### DISCUSSION TITLE
Multiple extends configuration

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#15488] [Suggest an Idea]
Hi,

I am using a self-hosted Renovate, early adopted in our company, where each team picks the configuration based on the package manager they are using. If someone is using docker, only extends the docker configuration, and if someone only is using node, only extends the node configuration. However, during the onboarding of a project that is using Node and Docker, renovate runs but only take into consideration the last extended configuration.

For example

This is the renovaje.json of a project
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "gitlab>***/renovate//config/docker/renovate",
        "gitlab>***/renovate//config/node16/renovate"
    ]
}
```
Where the `docker` configuration is 

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": ["dockerfile", "docker-compose"]
}
```

And the `node` configuration is 

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": ["npm", "nodenv", "nvm"],
  "updateInternalDeps": true,
  "skipInstalls": false,
}
```

When renovate runs, the `enabledManagers` of the last extended configuration is the only one that is taken into consideration. It seems something reasonable, but I wonder how could I approach to avoid this to happen.

We have multiple node, python, and other package managers, so we want to provide easy usage to the teams.

Thank you.

**Comment by rarkins** [comment]
There's no problem with multiple extends. The problem is that enabledManagers is not a mergeable configuration field, meaning last wins. You could maybe create a wrapper dockerNode extension which extends both and then adds enabledManagers

**Reply by juancarlosjr97** [reply]
Thank you very much for your reply @rarkins . I did not know `enabledManagers` was not a mergeable configuration field :( 

The wrapper looks like a good idea, thank you very much @rarkins

**Reply by Oliboy50** [reply]
ℹ️ I opened [this PR](https://github.com/renovatebot/renovate/pull/18603) to add it to the documentation
